parsers/UA.py

```diff
@@ -38,24 +39,27 @@
 
 def fetch_production(
     zone_key: str = "UA",
-    session: Optional[Session] = None,
     target_datetime: Optional[datetime] = None,
     logger: Logger = getLogger(__name__),
 ) -> list:
     if target_datetime:
-        raise NotImplementedError("This parser is not yet able to parse past dates")
-    r = session or Session()
+        target_date = arrow.get(target_datetime.date()).format("DD.MM.YYYY")
```

Comment:
Can you explain the first paragraph? Shouldn't we be expecting the exact time that we requested and only it? Logic for returning only one hour is at lines 91-97 and can be thrown out if we want it to return the whole day.

Reply:
We should be returning as much data as the source returns then we can set a re-fetch frequency so we only have to run it 1 per day we are re-fetching instead of 24.

Reply:
I changed it, so now parser will return the whole day of data (which is exactly what API gives us)